File: 344_reverse_string.rb
# @param {String} s
# @return {String}
def reverse_string(s)
  start = 0

  while start < s.length / 2
    swap_index = s.length - 1 - start
    s[start], s[swap_index] = s[swap_index], s[start]
    
    start += 1
  end

  s
end

p reverse_string("hello")

File: 344_reverse_string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseString = function (s) {
  let chars = s.split("");

  for (let i = 0; i < Math.floor(chars.length / 2); ++i) {
    [chars[i], chars[chars.length - i - 1]] = [chars[chars.length - i - 1], chars[i]];
  }

  return chars.join("");
};

// Strings in javascript are IMMUTABLE!

console.log(reverseString("hello"));